libmisc/malloc.c

```diff
@@ -20,27 +21,53 @@
 
 #ident "$Id$"
 
-#include <stdio.h>
+#include "malloc.h"
+
 #include <errno.h>
+#include <stddef.h>
+#include <stdio.h>
+
 #include "defines.h"
 #include "prototypes.h"
 #include "shadowlog.h"
 
-/*@maynotreturn@*/ /*@only@*//*@out@*//*@notnull@*/void *xmalloc (size_t size)
+
+extern inline void *xmalloc(size_t size);
+extern inline void *xmallocarray(size_t nmemb, size_t size);
+extern inline void *mallocarray(size_t nmemb, size_t size);
+extern inline void *reallocarrayf(void *p, size_t nmemb, size_t size);
+extern inline char *xstrdup(const char *str);
+
+
+void *
+xcalloc(size_t nmemb, size_t size)
 {
-	void *ptr;
-
-	ptr = malloc (size);
-	if (NULL == ptr) {
-		(void) fprintf (log_get_logfd(),
-		                _("%s: failed to allocate memory: %s\n"),
-		                log_get_progname(), strerror (errno));
-		exit (13);
-	}
-	return ptr;
+	void  *p;
+
+	p = calloc(nmemb, size);
+	if (p == NULL)
+		goto x;
+
+	return p;
+
+x:
+	fprintf(log_get_logfd(), _("%s: %s\n"),
+	        log_get_progname(), strerror(errno));
+	exit(13);
 }
 
-/*@maynotreturn@*/ /*@only@*//*@notnull@*/char *xstrdup (const char *str)
+
+void *
+xreallocarray(void *p, size_t nmemb, size_t size)
 {
-	return strcpy (xmalloc (strlen (str) + 1), str);
+	p = reallocarrayf(p, nmemb, size);
+	if (p == NULL)
+		goto x;
```

Comment:
Rename `x` to something more meaningful, i.e. done, end, fail...

Reply:
so that's the reason? In that case I withdraw my comment

Reply:
To be more precise, `xmalloc()` has been called that way historically (probably) because it was MALLOC or eXit (on failure).  The x comes from exit(3), and corresponds to the error path.  So yes, naming the label `x` makes sense in my head.